cmake/CMakeLists.txt

```diff
@@ -192,11 +190,6 @@ else()
     string(APPEND CMAKE_CXX_FLAGS_RELWITHDEBINFO " -march=native -mtune=native")
     string(APPEND CMAKE_C_FLAGS_RELWITHDEBINFO " -march=native -mtune=native")
   endif()
-  if(onnxruntime_BUILD_x86)
-    set (CMAKE_SYSTEM_PROCESSOR "x86")
-    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -msse2 -mfpmath=sse -Wno-narrowing")
```

Comment:
The reason these flags were here was so that the entire project, especially the providers, were built with SSE2 support enabled. The default for gcc is to fallback to legacy x87 FPU support. You should check something like the float32 Add operator and verify that Eigen is vectorizing (you should see addps/addss). As it is, I think only MLAS has SSE2 turned on.

For ARM, the MLAS cmake file has a switch for -mfpu=neon. Technically, this should also be at the global level. The dumpmachine check maybe should be at the global level and set some parameter that the MLAS cmake can use to select the right files.